lib/rack/content_type_default.rb

```diff
@@ -1,32 +1,54 @@
 # This is a simple Rack Middleware to set the request content type for specific routes. It checks if a content type is
-# set and does not override the existing one. The options allow you to specify the request methods, and the content type
-# to be set.
+# set and does not override the existing one. The options allow you to specify the request methods and the content type
+# to be set. The options also allow you to set a default content type only on select paths. In addition, the options
+# allow  you to set content type to to application/json or application/xml if possible based on the path ending.
 # Copied and adapted from https://gist.github.com/tstachl/6264249
-
 module Rack
   class ContentTypeDefault
-    def initialize(app, methods = [:post], content_type = 'application/json')
+    def initialize(app, methods = [:post], content_type = 'application/json', paths = 'all', default_based_on_path = false)
       @app = app
       @methods = [*methods].map{ |item| item.to_s.upcase }
       @content_type = content_type
+      @paths = [*paths]
+      @default_based_on_path = default_based_on_path
```

Comment:
how about `infer_based_on_path` for this variable rather than the word `default_`, or something shorter? can't think of anything as of now. basically we are giving it a hint. how's `use_path_hint`?